src/libraries/System.IO.Pipes/src/System/IO/Pipes/NamedPipeServerStream.Unix.cs

```diff
@@ -325,6 +328,11 @@ private SharedServer(string path, int maxCount)
                     socket.Bind(new UnixDomainSocketEndPoint(path));
                     socket.Listen(int.MaxValue);
                 }
+                catch (SocketException) when (isFirstPipeInstance)
```

Comment:
Apologies, my suggestion was incomplete. After checking [Socket.Listen](https://learn.microsoft.com/pl-pl/dotnet/api/system.net.sockets.socket.listen?view=net-7.0) I've realized that it can throw `SocketException` as well. So we need to catch it only when thrown by `Socket.Bind` (assuming that the comment above is correct).

We could use a nested try-catch block:

```cs
var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
try
{
    try
    {
        socket.Bind(new UnixDomainSocketEndPoint(path));
    }
    catch (SocketException) when (isFirstPipeInstance)
    {
        throw new UnauthorizedAccessException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, path));
    }   
    
    socket.Listen(int.MaxValue);
}
catch
{
    socket.Dispose();
    throw;
}
```

But we can also just use a boolean flag:

```cs
var socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.Unspecified);
bool bound = false;
try
{
    socket.Bind(new UnixDomainSocketEndPoint(path));
    bound = true;
    socket.Listen(int.MaxValue);
}
catch (SocketException) when (!bound && isFirstPipeInstance)
{
    socket.Dispose();
    throw new UnauthorizedAccessException(SR.Format(SR.UnauthorizedAccess_IODenied_Path, path));
}
catch
{
    socket.Dispose();
    throw;
}
```

Reply:
I went with second implementation suggested as it's nested in a better way